x-pack/plugins/task_manager/server/lib/get_next_run_at.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { intervalFromDate } from './intervals';
+import type { ConcreteTaskInstance } from '../task';
+
+export function getNextRunAt(
+  task: ConcreteTaskInstance,
+  taskUpdates: Partial<Pick<ConcreteTaskInstance, 'runAt' | 'schedule'>>,
+  taskDelayThresholdForPreciseScheduling: number = 0
+): Date {
+  const { runAt: originalRunAt, startedAt, schedule } = task;
+  const { runAt: newRunAt, schedule: newSchedule } = taskUpdates;
+
+  if (newRunAt) {
+    return newRunAt;
+  }
+
+  const taskSchedule = newSchedule?.interval ?? schedule?.interval;
```

Comment:
nit. since this is also calculated in the task runner, we could set `const scheduleToUse = newSchedule ?? schedule` in the task runner and pass it into the function

Reply:
@ymao1 Something like this may work, lmk what you think! https://github.com/elastic/kibana/pull/190093/commits/a396408f06f7bffe4779c5df4f58410173108c6b

Reply:
LGTM